python/lsst/ts/atspectrograph/atspec_csc.py

```diff
@@ -102,26 +102,64 @@ def __init__(self, config_dir=None, initial_state=salobj.State.STANDBY,
 
             self.want_connection = False
 
-        for query, report_state in [("query_fw_status", "fwState"),
-                                    ("query_gw_status", "gwState"),
-                                    ("query_gs_status", "lsState")]:
-            try:
-                state = await getattr(self.model, query)(self.want_connection)
-                self.log.debug(f"{query}: {state}")
-                getattr(self, f"evt_{report_state}").set_put(state=state[0])
-            except Exception as e:
-                self.evt_errorCode.set_put(errorCode=CONNECTION_ERROR,
-                                           errorReport=f"Cannot get information from model for "
-                                                       f"{report_state}.",
-                                           traceback=traceback.format_exc())
-                self.fault()
-                raise e
+        try:
+            # Check/Report linear stage position. Home if position is out of
+            # range.
+            state = await self.model.query_gs_status(self.want_connection)
+            self.log.debug(f"query_gs_status: {state}")
+            if state[1] < 0.:
+                self.log.warning("Linear stage out of range. Homing.")
+                await self.home_element(query="query_gs_status",
+                                        home="init_gs",
+                                        report="reportedLinearStagePosition",
+                                        inposition="linearStageInPosition",
+                                        report_state="lsState")
+            else:
+                self.evt_reportedLinearStagePosition.set_put(position=state[1])
+        except Exception as e:
+            self.evt_errorCode.set_put(errorCode=CONNECTION_ERROR,
+                                       errorReport=f"Cannot get information from model for "
+                                                   f"linear stage.",
+                                       traceback=traceback.format_exc())
+            self.fault()
```

Comment:
Please replace 120-124 with:
```
self.fault(code=CONNECTION_ERROR,
            report=f"Cannot get information from model for linear stage.",
            traceback=traceback.format_exc())
```
And similarly everywhere that you call fault().

Calling fault() with no arguments is deprecated and triggers a deprecation warning, which you should see when you run unit tests.

Reply:
nice!